src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -0,0 +1,319 @@
+     * By creating it you are getting the register from the pool RegistersPool.
+     * It could be created by using constructor with RegistersPool as an argument, like the next:
+     *      const RegistersPool::Reg<Xbyak::Reg64> reg {regPool};
+     * The destructor will return the register to the pool. Or it could be returned manually:
+     *      reg.release();
+     * @tparam TReg Xbyak register class
+     */
+    template<typename TReg>
+    class Reg {
+        friend class RegistersPool;
+    public:
+        Reg() {}
+        Reg(const RegistersPool::Ptr& regPool) { initialize(regPool); }
+        Reg(const RegistersPool::Ptr& regPool, int requestedIdx) { initialize(regPool, requestedIdx); }
+        ~Reg() { release(); }
+        Reg& operator=(Reg&& other)  noexcept {
+            release();
+            reg = other.reg;
+            regPool = std::move(other.regPool);
+            return *this;
+        }
+        Reg(Reg&& other)  noexcept : reg(other.reg), regPool(std::move(other.regPool)) {}
+        operator TReg&() { ensureValid(); return reg; }
+        operator const TReg&() const { ensureValid(); return reg; }
+        operator Xbyak::RegExp() const { ensureValid(); return reg; }
+        int getIdx() const { ensureValid(); return reg.getIdx(); }
+        friend Xbyak::RegExp operator+(const Reg& lhs, const Xbyak::RegExp& rhs) {
+            lhs.ensureValid();
+            return lhs.operator Xbyak::RegExp() + rhs;
+        }
+        void release() {
+            if (regPool) {
+                regPool->returnToPool(reg);
+                regPool.reset();
+            }
+        }
+        bool isInitialized() const { return static_cast<bool>(regPool); }
+
+    private:
+        void ensureValid() const {
+            if (!isInitialized()) {
+                IE_THROW() << "RegistersPool::Reg is either not initialized or released";
+            }
+        }
+
+        void initialize(const RegistersPool::Ptr& pool, int requestedIdx = anyIdx) {
+            static_assert(ov::util::IsAnyOf<TReg, Xbyak::Xmm, Xbyak::Ymm, Xbyak::Zmm,
+                                  Xbyak::Reg8, Xbyak::Reg16, Xbyak::Reg32, Xbyak::Reg64, Xbyak::Opmask>::value,
+                          "Unsupported TReg by RegistersPool::Reg. Please, use the following Xbyak registers either "
+                          "Reg8, Reg16, Reg32, Reg64, Xmm, Ymm, Zmm or Opmask");
+            release();
+            reg = TReg(pool->template getFree<TReg>(requestedIdx));
+            regPool = pool;
+        }
+
+    private:
+        TReg reg;
+        RegistersPool::Ptr regPool;
+    };
+
+    virtual ~RegistersPool() {
+        checkUniqueAndUpdate(false);
+    }
+
+    template <x64::cpu_isa_t isa>
+    static Ptr create(std::initializer_list<Xbyak::Reg> regsToExclude);
+
+    template<typename TReg>
+    size_t countFree() const {
+        static_assert(ov::util::IsAnyOf<TReg, Xbyak::Xmm, Xbyak::Ymm, Xbyak::Zmm,
```

Comment:
This static_assert could be more simpler:
```cpp
static_assert(std::is_base_of<Xbyak::Reg, TReg>::value && !std::is_base_of<Xbyak::Mmx, TReg>::value, "...");
```

Reply:
Actually, all Xbyak registers are derived from Xbyak::Reg, but there are Xbyak::BoundsReg and Xbyak::Fpu also, which we didn't use in the RegistersPool.
![Screenshot from 2022-09-21 10-39-56](https://user-images.githubusercontent.com/95029036/191445356-eca9bcc5-a767-44c4-846e-05b359063179.png)